sig-release/members.md

```diff
@@ -0,0 +1,8 @@
+## SIG Release Membership
```

Comment:
@calebamiles do we need to count the heads of members? I've been speaking about the people with the leadership role.